core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2439,11 +2439,17 @@ private[spark] object Utils
 
   /**
    * Returns the current user name. This is the currently logged in user, unless that's been
-   * overridden by the `SPARK_USER` environment variable.
+   * overridden by the `SPARK_USER` environment variable. In case of exceptions, returns 'spark'.
    */
   def getCurrentUserName(): String = {
-    Option(System.getenv("SPARK_USER"))
-      .getOrElse(UserGroupInformation.getCurrentUser().getShortUserName())
+    try {
+      Option(System.getenv("SPARK_USER"))
+        .getOrElse(UserGroupInformation.getCurrentUser().getShortUserName())
+    } catch {
+      // JEP 486: Permanently Disable the Security Manager
+      case e: UnsupportedOperationException if e.getMessage().contains("getSubject") =>
+        "spark"
```

Comment:
is it safer to fallback to the JVM property `user.name`?

Reply:
Thank you. Sure, that sounds like a reasonable default value, @yaooqinn .
Let me make a follow-up.

Reply:
Thank you @dongjoon-hyun